src/extension/postgres/extension.rs

```diff
@@ -0,0 +1,287 @@
+use crate::SqlWriter;
+
+#[derive(Debug, Default, Clone, PartialEq, Eq)]
+pub(crate) enum ExtensionOperation {
+    #[default]
+    Create,
+    Drop,
+}
+
+/// Creates a new "CREATE EXTENSION" statement for PostgreSQL
+///
+/// # Examples
+///
+/// ```
+/// use sea_query::extension::postgres::{ExtensionBuilder, ExtensionStatement};
+/// use sea_query::tests_cfg::*;
+///
+/// let mut query = String::new();
+/// let stmt = ExtensionStatement::create("ltree")
+///     .if_not_exists()
+///     .cascade()
+///     .schema("public")
+///     .version("v0.1.0")
+///     .to_owned();
+///
+/// stmt.prepare_extension_statement(&mut query);
+///
+/// assert_eq!(
+///     query,
+///     r#"CREATE EXTENSION IF NOT EXISTS ltree WITH SCHEMA public VERSION v0.1.0 CASCADE"#
+/// );
+/// ```
+///
+/// # References
+///
+/// [Refer to the PostgreSQL Documentation][1]
+///
+/// [1]: https://www.postgresql.org/docs/current/sql-createextension.html
+#[derive(Debug, Default, Clone, PartialEq, Eq)]
+pub struct ExtensionStatement {
+    pub(crate) name: String,
+    pub(crate) schema: Option<String>,
+    pub(crate) version: Option<String>,
+    /// Conditional to execute query based on existance of the extension.
+    ///
+    /// This is only used for `CREATE EXTENSION` and is not compatible with
+    /// `DROP EXTENSION`.
+    pub(crate) if_not_exists: bool,
+    /// Conditional to execute query based on existance of the extension.
+    ///
+    /// This is only used for `DROP EXTENSION` and is not compatible with
+    /// `CREATE EXTENSION`.
+    pub(crate) if_exists: bool,
+    /// Determines the presence of the `RESTRICT` statement.
+    ///
+    /// This is only used for `DROP EXTENSION` and is not compatible with
+    /// `CREATE EXTENSION`.
+    pub(crate) restrict: bool,
+    /// Determines the presence of the `RESTRICT` statement
+    pub(crate) cascade: bool,
+    pub(crate) operation: ExtensionOperation,
+}
+
+pub trait ExtensionBuilder {
+    /// Translate [`ExtensionStatement`] into a PostgreSQL's `CREATE` or `DROP`
+    /// `EXTENSION` statement.
+    ///
+    /// ## Refer
+    ///
+    /// https://www.postgresql.org/docs/current/sql-createextension.html
+    fn prepare_extension_statement(&self, sql: &mut dyn SqlWriter);
+}
```

Comment:
Hey @EstebanBorai, sorry, to clarify. I mean the `ExtensionBuilder` should have similar methods as the type statement below. I understand that the `ExtensionStatement ` is only available in Postgres but we want to keep the API looks unify.

https://github.com/SeaQL/sea-query/blob/9ff95dcfeeec8b99afad1ec6474676c92163c0a6/src/extension/postgres/types.rs#L466-L500

Reply:
Hi @billy1624! Thanks for clarifying, I just updated the PR following the specifcations.
Let me know your feedback please!